Lib/contextlib.py

```diff
@@ -31,6 +32,28 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+class AbstractAsyncContextManager(abc.ABC):
+
+    """An abstract base class for asynchronous context managers."""
+
+    async def __aenter__(self):
+        """Return `self` upon entering the runtime context."""
+        return self
+
+    @abc.abstractmethod
+    async def __aexit__(self, exc_type, exc_value, traceback):
+        """Raise any exception triggered within the runtime context."""
+        return None
+
+    @classmethod
+    def __subclasshook__(cls, C):
+        if cls is AbstractAsyncContextManager:
+            if (any("__aenter__" in B.__dict__ for B in C.__mro__) and
+                any("__aexit__" in B.__dict__ for B in C.__mro__)):
```

Comment:
I understand you just copied this from above, but I think both should be updated to support the explicit "anti-registration" ``__enter__ = None`` pattern, like all classes in ``collections.abc``, see ``_check_methods`` function in ``_collections_abc.py``.